### ISSUE TITLE
loss does note decrese when training resnet 50 

### ISSUE THREAD

**Issue by creativesh** [BVLC/caffe#6050]
hi 
I am training a resNet 50 network from scratch on audioset 2017 ,
my framework in tensorflow,but after some iteration,loss doesnt change 
my batchsize is 128 due to paper 
initial vale of learning rate is 1e-4,i have deacreased and increased it many times,but it has no effect.
this is my model implementation:
    with tf.variable_scope('conv_layer1'):
      net = slim.conv2d(
            x,
            num_outputs=64,
            kernel_size=7,
            stride=1,
            activation_fn=None,
            biases_initializer=tf.zeros_initializer()
      )
      net = slim.batch_norm(net)
      net=tf.nn.relu(net)
        # Max pool
    net = slim.max_pool2d(
        net, kernel_size=[3, 3], stride=2, padding='same')

    with tf.variable_scope('net_shortcut'):
        net_shortcut = slim.conv2d(
            net,
            num_outputs=groups[0].num_filters,
            kernel_size=1,
            stride=1,
            activation_fn=None,
            biases_initializer=None
        )
        net_shortcut = slim.batch_norm(net_shortcut)

    # Create the bottleneck groups, each of which contains `num_blocks`
    # bottleneck groups.
    for group_i, group in enumerate(groups):
        for block_i in range(group.num_blocks):
            name = 'group_%d/block_%d' % (group_i, block_i)
            first_strid = 1 if (group_i == 0 or block_i > 0) else 2
            # 1x1 convolution responsible for reducing dimension
            with tf.variable_scope(name + '/conv_in'):
                conv = slim.conv2d(
                        net,
                        num_outputs=group.num_filters / 4,
                        kernel_size=1,
                        stride=first_strid,
                        activation_fn=None,
                        biases_initializer=None
                )
                conv = slim.batch_norm(conv)
                conv = tf.nn.relu(conv)

            with tf.variable_scope(name + '/conv_bottleneck'):
                conv = slim.conv2d(
                        conv,
                        num_outputs=group.bottleneck_size,
                        kernel_size=3,
                        #padding='same',
                        activation_fn=None,
                        biases_initializer=None
                )
                conv = slim.batch_norm(conv)
                conv=tf.nn.relu(conv)

            # 1x1 convolution responsible for restoring dimension
            with tf.variable_scope(name + '/conv_out'):
                input_dim = net_shortcut.get_shape()[-1].value
                conv = slim.conv2d(
                        conv,
                        num_outputs=input_dim,
                        kernel_size=1,
                        activation_fn=None,
                        biases_initializer=None
                        )
                conv = slim.batch_norm(conv)
            # shortcut connections that turn the network into its counterpart
            # residual function (identity shortcut)
            with tf.variable_scope(name + '_add_relue'):
                if block_i == 0:
                    net = tf.nn.relu(conv + net_shortcut)
                else:
                    net = tf.nn.relu(conv + net)

        try:
            # upscale to the next group size
            next_group = groups[group_i + 1]
            with tf.variable_scope('group_%d/conv_upscale' % group_i):
                net_shortcut = slim.conv2d(
                    net,
                    num_outputs=next_group.num_filters,
                    kernel_size=1,
                    stride=2,
                   # padding='same',
                    activation_fn=None
                )
                net_shortcut = slim.batch_norm(net_shortcut)
        except IndexError:
                pass
    net = tf.nn.avg_pool(
        net,
        ksize=[1, 6, 4, 1],
        strides=[1, 1, 1, 1],
        padding='VALID')
    net = slim.flatten(net)
    net = slim.fully_connected(net, 1000, scope='fc2')#num_classes=527
    return tf.identity(net,name='net_out')

**Comment by shaibagon** [comment]
Please use the [caffe-users list][1] for usage, installation, or modeling questions, or other requests for help.
You may also post questions on [stackoverflow][3], make sure you tag them with `caffe` tag.  
There is also [caffe.help][4] documenting the different layers of caffe.
Do not post such requests to Issues. Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing][2] before submitting this issue.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
 [3]: http://stackoverflow.com/questions/tagged/caffe
 [4]: http://caffe.help/manual/layers.html
